### PR TITLE
T7.2

### PR DESCRIPTION
**Pull request by olikami**

### CODE REVIEW

File: mlClassifier/app.py [hunk 1]
```diff
@@ -1,14 +1,104 @@
...
+from flask import Flask, Response, request
+import os, werkzeug
+import requests
 
 app = Flask(__name__)
 
+types = [PYTHON3_LANG_NAME, KOTLIN_LANG_NAME, JAVA_LANG_NAME, "python"]
+
+# Helper method to make testing easier
+def getLexing(text, type):
```
Comment (nikohlUZH) [inline mlClassifier/app.py:13]: I took the highlight and lexing call out of the functions below. This makes mocking them for easier. @alainkueng @epura

File: mlClassifier/app.py [hunk 2]
```diff
@@ -1,14 +1,104 @@
...
+
+# Helper method to make testing easier
+def getHighlightString(text, type):
+    return requests.get(
+            "http://localhost:8080/highlight-string",
+            # for docker: http://formalSyntaxHighlighter:8080/highlight-string
+            params={"text": text, "type": type},
+        )
+
+def configure_routes(app):
```
Comment (nikohlUZH) [inline mlClassifier/app.py:27]: I wrapped the two functions into configure_routes @alainkueng @epura

File: mlClassifier/app.py [hunk 3]
```diff
@@ -1,14 +1,104 @@
...
+        if type not in types:
+            return "The type should be either 'java', 'kotlin', 'python'!", 400
+        if not isinstance(text, str):
+            return "Text should be a string!", 400
+
+        # convert 'python3' variable to python for the formalSyntaxHighlighter API call
+        if type == PYTHON3_LANG_NAME:
+            type = 'python'
+        formal_syntax_highlighter = getLexing(text, type)
+        fsh_response = formal_syntax_highlighter.json
```
Comment (nikohlUZH) [inline mlClassifier/app.py:45]: This used to be: fsh_response = formal_syntax_highlighter.json() .... I consistently got an error telling me that a dict object is not callable, hence this change. @epura @alainkueng

File: mlClassifier/app.py [hunk 4]
```diff
@@ -1,14 +1,104 @@
...
+        if type not in types:
+            return "The type should be either 'java', 'kotlin', 'python'!", 400
+
+        if not isinstance(text, str):
+            return "Text should be a string!", 400
+
+        if type == PYTHON3_LANG_NAME:
+            type = 'python'
+
+        formal_syntax_highlighter = getHighlightString(text, type)
```
Comment (nikohlUZH) [inline mlClassifier/app.py:82]: same as above: This used to be: fsh_response = formal_syntax_highlighter.json() .... I consistently got an error telling me that a dict object is not callable, hence this change. @epura @alainkueng

**Review by nikohlUZH — COMMENTED** [review]
I had to make some changes to the app.py file. Can you please check if these changes are ok for you?

**Comment by Chreggii** [comment]
@olikami what is the reason for replacing all single quotes with double quotes? I prefer the single quotes, since you can nest double quotes in single quotes but not the other way around (typescript).

**Comment by olikami** [comment]
> @olikami what is the reason for replacing all single quotes with double quotes? I prefer the single quotes, since you can nest double quotes in single quotes but not the other way around (typescript).

No particular reason in this case, just because it was the default that was included. My personal reasons for double quotes are:

- Similarity to C language
- JSON
- I use more apostrophes than quote marks (at least in English)

As far as I know, you can very much nest single quotes in double quotes as well as double quotes in single quotes.

I think we should discuss this in the next meeting, come up with a consistent style, and then enforce it.

**Review by nikohlUZH — APPROVED** [review]
Looks good to me :) well done everyone

**Pull request merged**